ui/dashboards/src/context/DashboardProvider/DashboardProvider.tsx

```diff
@@ -78,27 +89,19 @@ export function DashboardProvider(props: DashboardProviderProps) {
           ...createDeletePanelGroupSlice(...args),
           ...createPanelEditorSlice()(...args),
           ...createDeletePanelSlice()(...args),
-          defaultTimeRange: { pastDuration: dashboardSpec.duration },
+          metadata,
+          defaultTimeRange: { pastDuration: duration },
           isEditMode: !!isEditMode,
           setEditMode: (isEditMode: boolean) => set({ isEditMode }),
-          reset: () => {
-            const { resetPanels, resetPanelGroups } = get();
-            resetPanels();
-            resetPanelGroups();
-          },
-          save: () => {
-            const { savePanels, savePanelGroups } = get();
-            savePanels();
-            savePanelGroups();
+          reset: ({ spec: { panels, layouts } }) => {
+            const { setPanels, setPanelGroups } = get();
```

Comment:
Do we actually need these as "public" methods on the store? Feels like reset here should just be able to call `set` on whatever state it wants since it's composed of the other slices?

Basically, if these `set` methods aren't actually called by any UI components directly, would be nice if they weren't part of the public API of the store.

Reply:
oh good point we can set the panels and groups here directly